13-ngrx-effects.md

```diff
@@ -22,14 +25,14 @@ export class AuthEffects {
   @Effect()
   login$ = this.actions$.pipe(
     ofType(AuthActionTypes.Login),
-    mergeMap((action: authActions.Login) =>
-      this.authService
-        .login(action.payload)
-        .pipe(
-          map((user: User) => new authActions.LoginSuccess(user)),
-          catchError(error => of(new authActions.LoginFail(error)))
-        )
-    )
+    fetch({
```

Comment:
Where does the fetch symbol come from is there an import?

Is this the code from nx to handle and effect? I have always stayed NgRx focused and not used the Nx stuff as it likely will not be mass adopted.

Saying that I am totally open to teaching it but I wonder then should we have one of each style so the learner can decide?

Reply:
Using the generate command such as ng g nrgx products will create effects with this operator.  So I assumed that is the way it should be shown now.

The official docs say this in the Data Fetching section of the [nx docs](https://nx.dev/latest/angular/guides/misc-data-persistence):
*The fetch operator provides consistency when fetching data. If there are multiple requests scheduled for the same action, it will only run the last one.*

I can include the fetch import:

```js
import { fetch } from '@nrwl/angular';
```

Looking at the codebase at work, fetch is not used anywhere.  So I think you have a good point.  However, if we are using a generate command for NgRx, then maybe a description can be added noting the different scaffolding commands, instead of a different file with the nx command shown.

Another issue is that the User class would also not be needed if this:

```js
return AuthActions.loginSuccess(action);
```

Really replaces this:

```js
map((user: User) => new authActions.LoginSuccess(user)),
```

Let me know how you want to handle this.

Reply:
I am leaning to showing ngrx as most people use it. But happy to have an example of both for the first effect then just use one type for the rest

What do you think?

Reply:
Sorry for the delay on this.  Regarding showing older versions of the code, I guess having the original and then the latest versions is OK.  I personally think that whatever is shown in the official code snippets is the way to go.  However, when I first conceived the idea to update this workshop as a valuable example of enterprise NgRx, I did have both old an new versions described. 
However, I think NgRx will continually get more concise and follow the same direction that Redux is going in with React Hooks.  Using createAction, createEffect and createReducer with the 'on' syntax instead of a switch statement is pretty important.
Anyhow, it's worth seeing how having both legacy and current code examples side by side might look.  I will give an attempt of this for the section 14/15.
Also, it seems to me that section 14 should be in section 15 *after* the products lib is created.  I would like to introduce the product modules scaffolding command in section 14, talking about a selector there, and then working on the rest of the product module in section 15.  At least that's what I had underway.  At work they seem to be moving away from using NgRx as developers on other projects have resisted it, and new projects going forward are just using a kind of observable state pattern.
I will update this PR to reflect both code samples first.

Reply:
Hi,
My turn to apologise for the delay.

I think we should try use one style where possible. I know in the redo of my NgRx course on Pluralsight we went with only new syntax using on.

Have at moving the order around.

I have not got much time to contribute to changes of this repo at the moment but happy to review and add any insights or bounce ideas. If you need to chat more about it feel free to ask more questions or arrange a time to catch up.